### PR TITLE
add request string logging to be able to get more information

### PR DESCRIPTION
**Pull request by nareshkakubal**
Add request logging to get more insight when there is a failure

### CODE REVIEW

File: banyan/resource_accesstier.go [hunk 1]
```diff
@@ -23,11 +24,31 @@ func resourceAccessTier() *schema.Resource {
...
+	parts := strings.Split(inID, ":")
+	if len(parts) == 1 {
+		data.SetId(inID)
+		return []*schema.ResourceData{data}, nil
+	}
+	if len(parts) != 2 || strings.ToLower(parts[0]) != "name" {
+		return nil, fmt.Errorf("invalid ID (%s), expected name:<name of access tier to import>", parts)
+	}
+	name := parts[1]
+	accessTierClient := i.(*client.Holder)
```
Comment (yoshiotu) [inline banyan/resource_accesstier.go:43]: would it be good to check if the type assertion succeeded?

Comment (nareshkakubal) [reply banyan/resource_accesstier.go:43]: yes, sure shall add.

File: client/restclient/restclient.go [hunk 2]
```diff
@@ -235,8 +244,8 @@ func HandleResponse(response *http.Response, requestStr string) (responseData []
 		if err != nil {
 			return
 		}
-		uerr := json.Unmarshal(responseData, &errResp)
-		if uerr == nil {
+		myErr := json.Unmarshal(responseData, &errResp)
+		if myErr == nil {
 			err = fmt.Errorf("recieved error code %d: %s \n response: \n %s", response.StatusCode, requestStr, responseData)
```
Comment (yoshiotu) [inline client/restclient/restclient.go:249]: misspelling of "received"

Comment (yoshiotu) [reply client/restclient/restclient.go:249]: should this function also return an err here when `myErr` is non-nil?

Comment (nareshkakubal) [reply client/restclient/restclient.go:249]: i had same thought when i looked into, but looks like idea of the HandleResponse function was to print corresponding errors based on http error code, so when there are parsing errors it is left to the caller to handle. in this above case we need to set err to myErr.

File: client/restclient/restclient.go [hunk 3]
```diff
@@ -235,8 +244,8 @@ func HandleResponse(response *http.Response, requestStr string) (responseData []
 		if err != nil {
```
Comment (yoshiotu) [inline client/restclient/restclient.go]: This condition is never true

File: client/restclient/restclient.go (hunk 2 above)
Comment (yoshiotu) [reply client/restclient/restclient.go]: it's not doing returning or handling `myErr` at all currently.

File: client/restclient/restclient.go (hunk 3 above)
Comment (nareshkakubal) [reply client/restclient/restclient.go]: yes so removed

Comment (nareshkakubal) [reply client/restclient/restclient.go]: https://github.com/banyansecurity/terraform-provider-banyan/pull/155/commits/afa521abd870a6516d9b06b4d137b7bd9a8d066f#diff-5ec9649c5426a6d1aac77b9e75d2068d03eadd6e98abaa3a16100dfb9784d3fc

File: client/restclient/restclient.go (hunk 2 above)
Comment (nareshkakubal) [reply client/restclient/restclient.go]: yes i saw it, made corresponding change.

File: client/accesstier/client.go [hunk 4]
```diff
@@ -53,16 +53,33 @@ func (a *AccessTier) GetName(name string) (spec AccessTierInfo, err error) {
...
 	}
-	if j.Count == 0 {
+	if j.Data.Count == 0 {
 		err = fmt.Errorf("access tier with name %s not found", name)
 		return
 	}
-	return j.Data, nil
+	for _, accessTier := range j.Data.AccessTiers {
+		if accessTier.Name == name {
+			spec = accessTier
```
[the remark this answers is not in the document]
Comment (nareshkakubal) [reply client/accesstier/client.go:76]: yes correct shall add

**Review by yoshiotu — APPROVED** [review]

**Pull request merged**